src/A2A.AspNetCore/A2AEndpointRouteBuilderExtensions.cs

```diff
@@ -46,22 +46,49 @@ public static IEndpointConventionBuilder MapA2A(this IEndpointRouteBuilder endpo
     /// Enables the well-known agent card endpoint for agent discovery.
     /// </summary>
     /// <param name="endpoints">The endpoint route builder to configure.</param>
-    /// <param name="taskManager">The task manager for handling A2A operations.</param>
+    /// <param name="agentCard">The <see cref="AgentCard"/> to be configured.</param>
     /// <param name="agentPath">The base path where the A2A agent is hosted.</param>
     /// <returns>An endpoint convention builder for further configuration.</returns>
-    public static IEndpointConventionBuilder MapWellKnownAgentCard(this IEndpointRouteBuilder endpoints, ITaskManager taskManager, [StringSyntax("Route")] string agentPath)
+    public static IEndpointConventionBuilder MapWellKnownAgentCard(this IEndpointRouteBuilder endpoints, AgentCard agentCard, [StringSyntax("Route")] string agentPath)
```

Comment:
I added new overload for the ease of use: most users don't need to customize `AgentCardProvider` behaviour, so all they need to do is to provide a valid `AgentCard` here.

Reply:
Consider the case where the agent card is created via some calls to remote endpoints e.g. populate the skills by asking MCP servers for tools. This would require (ideally) async calls but since this method is utilized during DI and startup, it's non-trivial to make `async` calls to create the agent card to pass to this method.